### ISSUE TITLE
Feature Request Layout

### ISSUE THREAD

**Issue by Sypx** [NEZNAMY/TAB#615]
Make it possible to restrict number of columns, that is all I want and I believe many can benefit from, on the small server I am running, it is nice to know where everyone is at without covering the entire screen while doing so! I do not know how much of a hassle this is to develop but I believe that it will make the already amazing plugin more customizable than before and enable cleaner looking tab menus

**Comment by NEZNAMY** [comment]
Also feature requests are permanently closed.

**Comment by ColorsASD** [comment]
This is the biggest nonsense I've ever heard. The maximum number of columns is 80, right? If you limit it to just 40 or a smaller number, fewer columns will be created. You can't control the columns because vanilla dictates that for 20 players, it creates 1 or 2 columns. But we're already ahead if even just 2 columns appear, as the person above me mentioned.

80 slot:
![image](https://github.com/NEZNAMY/TAB/assets/12263156/978f5fc9-13f5-427d-8c02-1cad8acb1de9)

40 slot:
![image](https://github.com/NEZNAMY/TAB/assets/12263156/cc3f3e0b-9c7a-4eed-9814-10179f13c316)

**Comment by NEZNAMY** [comment]
Layout feature adds 80 fake slots and manages them. Using less than 80 means working with real players, which is much harder, since the plugin would also have to manage teams (for sorting), as well as a very detailed compatibility with plugins that add/remove players from tablist, including, but not limited to vanish plugins. It's just a lot of extra work and as mentioned above, feature requests are closed.

**Comment by ColorsASD** [comment]
You don't need to go into such detail. If you only add 40 fake players, the 41st player will be real. But if the server capacity is 40, it's extremely disgusting to see the unnecessary other 40 slots. From a design perspective, as I demonstrated earlier, it's simply laziness. Interesting how we managed to solve it. :)

**Comment by NEZNAMY** [comment]
I am not aware of a way to force the client to only display 40 slots. That's not laziness, unless you are referring to the missing feature, in which case I am legally allowed to not implement everything everyone in the world requests.

**Comment by Tanguygab** [comment]
When adding fake players to the tablist, real players still show as well at the end and can't be removed from it, otherwise they'd be invisible ingame. The only way we have found to bypass this issue is to fill all 80 slots that a normal Minecraft client can see, so it pushes real players outside of the visible list.

While 1.19.3 did add a way to hide player entries, TAB's layout is supported all the way back on 1.8, which means we cannot take advantage of this addition.

The only way to not display all 80 slots, as NEZ already said, is to manage both fake players and real players, and sort them together, which would mean:
- reworking the entire feature
- issues if the amount of real players goes over the amount of displayed columns
- not being able to hide players in the tablist through conditions while still seeing them in game

In short, this is not as simple as it seems and  just not worth our time.
